### PR TITLE
Add `WithContext` functions to executor

### PR DESCRIPTION
**Pull request by jmigueprieto**
## Changes in this PR

1. Added `WithContext` functions to executor e.g.: `StartWorkflowWithContext` for managing cancellations, timeouts, and deadlines.
2. Existing functions delegate calls to those new functions using `context.Background()`.
3. `HttpRequester` uses `http.NewRequestWithContext`.
4. Cleanup + small fixes.

---
```go

func (e *WorkflowExecutor) StartWorkflow(startWorkflowRequest *model.StartWorkflowRequest) (workflowId string, err error) {
	return e.StartWorkflowWithContext(context.Background(), startWorkflowRequest)
}

func (e *WorkflowExecutor) StartWorkflowWithContext(ctx context.Context, startWorkflowRequest *model.StartWorkflowRequest) (workflowId string, err error) {
	if err := ctx.Err(); err != nil {
		return "", err
	}

	id, _, err := e.workflowClient.StartWorkflowWithRequest(
		ctx,
		*startWorkflowRequest,
	)
	if err != nil {
		return "", err
	}
	return id, nil
}

```

### CODE REVIEW

File: sdk/model/rbac/create_or_update_application_request.go [hunk 1]
```diff
@@ -11,6 +11,5 @@ package rbac
 
 type CreateOrUpdateApplicationRequest struct {
 	// Application's name e.g.: Payment Processors
-	Name        string `json:"name"`
-	Description string `json"description"`
```
Comment (jmigueprieto) [inline sdk/model/rbac/create_or_update_application_request.go:15]: there is no description in `CreateOrUpdateApplicationRequest`

File: sdk/authentication/token_helper.go [hunk 2]
```diff
@@ -234,9 +234,7 @@ func contains(haystack []string, needle string) bool {
 }
 
 func setBody(body interface{}, contentType string) (bodyBuf *bytes.Buffer, err error) {
-	if bodyBuf == nil {
```
Comment (jmigueprieto) [inline sdk/authentication/token_helper.go:237]: unnecessary check

File: sdk/authentication/token_helper.go [hunk 3]
```diff
@@ -248,7 +246,7 @@ func setBody(body interface{}, contentType string) (bodyBuf *bytes.Buffer, err e
 	} else if jsonCheck.MatchString(contentType) {
 		err = json.NewEncoder(bodyBuf).Encode(body)
 	} else if xmlCheck.MatchString(contentType) {
-		xml.NewEncoder(bodyBuf).Encode(body)
+		err = xml.NewEncoder(bodyBuf).Encode(body)
```
Comment (jmigueprieto) [inline sdk/authentication/token_helper.go:249]: missing error handling but I don't think this is being used

File: sdk/client/http_requester.go [hunk 4]
```diff
@@ -139,12 +139,12 @@ func (h *HttpRequester) prepareRequest(
 	// Encode the parameters.
 	url.RawQuery = query.Encode()
 
-	// Generate a new request
 	if body != nil {
-		localVarRequest, err = http.NewRequest(method, url.String(), body)
+		localVarRequest, err = http.NewRequestWithContext(ctx, method, url.String(), body)
```
Comment (jmigueprieto) [inline sdk/client/http_requester.go:143]: context was propagated but not used at all before these changes

**Review by c4lm — APPROVED** [review]

File: test/integration_tests/executor_test.go [hunk 5]
```diff
@@ -90,3 +90,15 @@ func TestUpdate(t *testing.T) {
...
 }
+
+func TestStartWorkflowWithContext(t *testing.T) {
+	executor := testdata.WorkflowExecutor
+
+	ctx, cancel := context.WithCancel(context.Background())
+	// cancel straightaway on purpose
+	cancel()
+
+	_, err := executor.StartWorkflowWithContext(ctx, &model.StartWorkflowRequest{})
```
Comment (jmigueprieto) [inline test/integration_tests/executor_test.go:101]: An example test that shows that `StartWorkflowWithContext` throws an error if called with a cancelled context.

**Pull request merged**